egs/wsj/s5/utils/create_wdisambig_file.pl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/perl -w
+# Copyright 2016 FAU Erlangen (Author: Axel Horndasch)
+
+# usage:  create_wdisambig_file.pl category_list.txt words_or_phones.txt disambig_symbol_array.txt
+
+# This script creates a file which is usually called 'wdisambig.txt'. That file
+# contains a (line-by-line) list of the text-form of the disambiguation symbols
+# that are used in the grammar and passed through by the lexicon.
+#
+# See also 'prepare_lang.sh' (look for 'wdisambig.txt').
+use strict;
+use Getopt::Long;
+
+# options
+my $word_class_list     = "";
+my $remove_class_prefix = "";
+
+# helper variables
+my @word_list;
+
+my %optctl = (
+    word_class_list     => \$word_class_list,
+    remove_class_prefix => \$remove_class_prefix,
+    );
+
+# get the option values
+&GetOptions(\%optctl,
+    "word_class_list=s",
+    "remove_class_prefix=s",
+    );
+
+# start the list of disambiguation symbols with #0 (empty word symbol)
+print "#0\n";
+
+if(defined($word_class_list) and $word_class_list ne "") {
```

Comment:
All scripts have been revised to make them self-explanatory or at least much easier to understand. For that reason I changed the name of some of the scripts.